Alicerce/Sources/Logging/Chronicle/Log.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  Alicerce.swift
+//  Alicerce
+//
+//  Created by Meik Schutz on 03/04/17.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+final class Log {
+    
+    fileprivate static var providers = Array<LogProvider>()
+
+    static let defaultLevel: Level = Level.error
+    
+    // MARK:- Provider Management
+    
+    static var providerCount: Int {
+        return providers.count
+    }
+    
+    public class func registerProvider(provider: LogProvider) {
+        let matchingProviders = self.providers.filter { (registeredProvider) -> Bool in
+            return registeredProvider.providerInstanceId() == provider.providerInstanceId()
+        }
+        
+        if matchingProviders.count <= 0 {
+            self.providers.append(provider)
+        }
+    }
+    
+    public class func unregisterProvider(provider: LogProvider) {
+        self.providers = self.providers.filter({ (registeredProvider) -> Bool in
+            return registeredProvider.providerInstanceId() != provider.providerInstanceId()
+        })
+    }
+    
+    public class func removeAllProviders() {
+        self.providers.removeAll()
+    }
+    
+    // MARK:- Logging
+    
+    public class func verbose(message: String, file: String = #file, function: String = #function, line: Int = #line) {
+    
+        self.log(level: .verbose, message: message, file: file, function: function, line: line)
+    }
+
+    public class func debug(message: String, file: String = #file, function: String = #function, line: Int = #line) {
+        
+        self.log(level: .debug, message: message, file: file, function: function, line: line)
+    }
+
+    public class func info(message: String, file: String = #file, function: String = #function, line: Int = #line) {
+        
+        self.log(level: .info, message: message, file: file, function: function, line: line)
+    }
+
+    public class func warning(message: String, file: String = #file, function: String = #function, line: Int = #line) {
+        
+        self.log(level: .warning, message: message, file: file, function: function, line: line)
+    }
+
+    public class func error(message: String, file: String = #file, function: String = #function, line: Int = #line) {
+        
+        self.log(level: .error, message: message, file: file, function: function, line: line)
+    }
+    
+    public class func log(level: Level, message: String, file: String = #file, function: String = #function, line: Int = #line) {
+        
+        let item = LogItem(level: level, message: message, file: file, thread: self.threadName(), function: function, line: line)
+        for provider in self.providers {
+            if self.itemShouldBeLogged(provider: provider, item: item) {
```

Comment:
If you move this to the LogProvider, you'll need to implement the same logic into all provides, which I wanted to avoid. The minimum level filter should be equal to all providers, so that's why its in the Log class. Each LogProvided is free to do further filtering (in the ```write``` method) if so needed. Makes sense?